editor/plugins/canvas_item_editor_plugin.h

```diff
@@ -245,6 +246,9 @@ class CanvasItemEditor : public VBoxContainer {
 	bool key_rot;
 	bool key_scale;
 
+	bool is_snap_active() const { return snap_active ^ snap_key_enabled; }
```

Comment:
Honestly there's no need to create such function for something that is used only in one place. Please just use a boolean variable inside the snap_point and snap_angle to avoid bloating the class definition with single-use functions.

Reply:
It is also called in _draw_rulers

Reply:
Copy-paste the code then, it's one operation. The function calling is probably a lot more complex, performance-wise, than just a boolean operation.

Reply:
Its inline function :)

Reply:
Could have been one if you had added _FORCE_INLINE_ in front of the function. :)
Here it was just a function returning a const value.

Reply:
As far as I know all function in header files are inline by default

Reply:
Wow, that's actually true. You taught me something today ^^